charmhelpers/contrib/openstack/vaultlocker.py

```diff
@@ -54,27 +58,48 @@ def __call__(self):
 
                     # Tokens may change when secret_id's are being
                     # reissued - if so use token to get new secret_id
-                    if token != last_token:
+                    token_success = False
+                    try:
                         secret_id = retrieve_secret_id(
                             url=vault_url,
                             token=token
                         )
+                        token_success = True
+                    except hvac.exceptions.InvalidRequest:
+                        # Try next
+                        pass
+
+                    if token_success:
                         db.set('secret-id', secret_id)
-                        db.set('last-token', token)
                         db.flush()
 
-                    ctxt = {
-                        'vault_url': vault_url,
-                        'role_id': json.loads(role_id),
-                        'secret_id': secret_id,
-                        'secret_backend': self.secret_backend,
-                    }
-                    vault_ca = data.get('vault_ca')
-                    if vault_ca:
-                        ctxt['vault_ca'] = json.loads(vault_ca)
-                    self.complete = True
-                    return ctxt
-        return {}
+                        ctxt['vault_url'] = vault_url
+                        ctxt['role_id'] = json.loads(role_id)
+                        ctxt['secret_id'] = secret_id
```

Comment:
This secret-id needs to be persisted as its now the most recent one that's been retrieved.

Reply:
Oh wait I see that above now!